lib/facter/splunk_version.rb

```diff
@@ -1,16 +1,17 @@
 Facter.add(:splunk_version) do
   setcode do
     value = nil
-    cmd = if File.exist?('C:/Program Files/Splunk/bin/splunk.exe')
-            '"C:/Program Files/Splunk/bin/splunk.exe" --version'
-          elsif File.exist?('/opt/splunk/bin/splunk')
-            '/opt/splunk/bin/splunk --version'
-          end
-    if cmd
-      output = Facter::Util::Resolution.exec(cmd)
-      if output =~ %r{^Splunk ([0-9\.]+) \(} # rubocop:disable Style/IfUnlessModifier
-        value = Regexp.last_match(1)
-      end
+    kernel = Facter.value(:kernel)
+    version_file = case kernel
```

Comment:
Would it make sense to confine on version_file? I think that makes it more readable.